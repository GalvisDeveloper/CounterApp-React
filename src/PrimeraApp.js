// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre> { JSON.stringify(constante_objeto,null,3) } </pre> */}
            <p>{subtitulo}</p>
        </>
    );

}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Mi primera App'
}

export default PrimeraApp;
