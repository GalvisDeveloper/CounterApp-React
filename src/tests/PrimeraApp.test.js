

import React from 'react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

// import { render } from '@testing-library/react';
// import '@testing-library/jest-dom';

describe('Probando primeraApp', () => {

    // test('Debe mostrarse el saludo ', () => {

    //     const saludo = 'CounterApp';

    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect(getByText(saludo)).toBeInTheDocument();

    // })

    test('Debe mostrar <PrimeraApp /> Correctamente ', () => {

        const saludo = 'CounterApp';

        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();

    });


    test('Debe mostrar el subtitulo en <PrimeraApp /> Correctamente ', () => {

        const saludo = 'CounterApp';
        const subtitulo = 'Mi primera App';

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const saludoH1 = wrapper.find('h1').text();
        // const textoP = wrapper.find('p').text();
        // console.log(saludoH1);
        // console.log(textoP);
        expect(saludoH1).toBe(saludo);

    });



});