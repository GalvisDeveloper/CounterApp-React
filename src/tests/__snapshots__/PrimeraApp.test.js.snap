// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando primeraApp Debe mostrar <PrimeraApp /> Correctamente  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    Mi primera App
  </p>
</Fragment>
`;
